### ISSUE TITLE
validate_ami_bags.py throws formatting error for motion picture json files (film)

### ISSUE THREAD

**Issue by genfhk** [NYPL/ami-tools#59]
`check_techmd_fields` doesnt recognize film json; needs to allow for lack of audio codec (for motion picture film) or video codec (for audiofilm)

...

check `self.media_format_type = self.dict["source"]["object"]["type"][0:5]`

**Comment by genfhk** [comment]
if type is film, look for filmElement > audio

**Comment by bturkus** [comment]
could someone (@genfhk @nkrabben) confirm for me: this looks like a generic "WARNING - JSON metadata out of spec: JSON files contain formatting errors" message with no other detail?

**Comment by genfhk** [comment]
> 

Yeah - although I'd want to check your json file to be sure the error you're getting is for the same reasons mentioned in this issue. @nkrabben can you make time to resolve this issue sooner than later? It does make it difficult to know if there are actual other issues with json files...

**Comment by nkrabben** [comment]
I'm deep in troubleshooting this and think I got it, but running into some pymediainfo issues.

Can either of you check how pymediainfo works on Apple chips?

```python3 -c "from pymediainfo import MediaInfo; print(MediaInfo._get_library())"```

I'm getting a block of errors about a missing dylib

**Comment by bturkus** [comment]
i'm getting this:
```
(<CDLL 'libmediainfo.0.dylib', handle 204dc8c80 at 0x102991cd0>, 105553159471200, '22.03', (22, 3))
```
Mac Mini, M1, 2020, 12.3.1

**Comment by nkrabben** [comment]
Thanks!

That confirms that the problem is with my development environment, and not just a general pymediainfo issue. (still might be good to rethink pymediainfo and just shell out to mediainfo itself instead.

More to come.

**Comment by bturkus** [comment]
would love to transition away from pymediainfo generally, and thanks for investigating this!

**Comment by kieranjol** [comment]
Sorry for butting in, but I was always afraid of that extra pymediainfo layer and always subprocessed directly to mediainfo itself.
